clients/admin-ui/src/features/connection-type/connection-type.slice.ts

```diff
@@ -147,8 +135,11 @@ export const {
 } = connectionTypeApi;
 
 export const selectConnectionTypes = createSelector(
-  connectionTypeApi.endpoints.getAllConnectionTypes.select({
-    search: "",
-  }),
-  ({ data }) => data?.items ?? []
+  [
+    (RootState) => RootState,
+    connectionTypeApi.endpoints.getAllConnectionTypes.select({
+      search: "",
+    }),
+  ],
+  (RootState, { data }) => data?.items ?? []
```

Comment:
I had to do this change a few times which is a workaround for https://github.com/reduxjs/redux-toolkit/issues/2484

from what I understand, when code splitting, typescript doesn't understand that doing a `.endpoints.{something}.select()` is okay since it isn't part of the `RootState`. doing an initial `(RootState) => RootState` seems to make it okay though (though I think technically it's casting to `any` and I'm not sure what that is). other than that line, the following lines are all typed properly, so I think it should be okay, but this might be something we want to come back to in the future